src/components/Link.js

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { Link as USWDSLink } from "@trussworks/react-uswds";
+import { Link as GatsbyLink } from "gatsby";
+
+// Pass only activeClassName and partiallyActive to GatsbyLink
+// Pass only fileType and size to file downloads
+const Link = ({
+  children,
+  to,
+  activeClassName,
+  partiallyActive,
+  variant,
+  fileType,
+  size,
+  ...other
+}) => {
+  // Assumes that any internal link will start with exactly one slash, and that anything else is external.
+  const internal = /^\/(?!\/)/.test(to);
+
+  // Assumed file download link structure
+  const file = /\.[0-9a-z]+$/i.test(to);
+
+  if (internal) {
+    if (variant === "nav") {
+      // Navigation link
+      return (
+        <GatsbyLink
+          to={to}
+          className="usa-link usa-link--nav"
+          activeClassName={activeClassName}
+          partiallyActive={partiallyActive}
+          {...other}
+        >
+          {children}
+        </GatsbyLink>
+      );
+    }
+    if (file) {
```

Comment:
This is just coding style / preference, but line 38 could be "else if"

Reply:
You know what, looking at it now, it makes sense to include that for understandability. Push made.